core/src/Revolution/Formatter/modManagerDateFormatter.php

```diff
@@ -0,0 +1,246 @@
+     * @param string|int $value The value to transform (a Unix timestamp or mysql-format string)
+     * @param bool $useOffset Whether to use the offset time (system setting) in the date calculation
+     * @param string|null $emptyValue The text to show when the $value passed is empty
+     * @return string The formatted date
+     */
+    public function formatTime($value, bool $useOffset = false, ?string $emptyValue = null): string
+    {
+        return $this->format($value, $this->managerTimeFormat, $useOffset, $emptyValue);
+    }
+
+    /**
+     * Transforms a date/time-related value into the specified DateTime format showing
+     * both date and time, including an optional system-specified separator
+     * @param string|int $value The value to transform (a Unix timestamp or mysql-format string)
+     * @param bool $useOffset Whether to use the offset time (system setting) in the date calculation
+     * @param string|null $emptyValue The text to show when the $value passed is empty
+     * @return string The formatted date
+     */
+    public function formatDateTime($value, bool $useOffset = false, ?string $emptyValue = null): string
+    {
+        $managerDateTimeSeparator = $this->modx->getOption('manager_datetime_separator', null, ', ', true);
+        $format = $this->managerDateFormat . $managerDateTimeSeparator . $this->managerTimeFormat;
+
+        return $this->format($value, $format, $useOffset, $emptyValue);
+    }
+
+    /**
+     * Formats a Resource-related date when applicable and includes localized default text to
+     * represent conditions where a date is not present or applicable
+     *
+     * @param string|int $value The source date value to format
+     * @param string $whichDate An identifier specifying the type of date being formatted
+     * @param bool $useStandardEmptyValue Whether to use the default empty value defined in the system settings
+     * @return string The formatted date or relevant text indicating an empty date value
+     */
+    public function formatResourceDate($value, string $whichDate = 'created', bool $useStandardEmptyValue = true): string
+    {
+        if ($useStandardEmptyValue) {
+            $emptyValue = null;
+        } else {
+            switch ($whichDate) {
+                case 'edited':
+                    $emptyValue = $this->modx->lexicon('unedited');
+                    break;
+                case 'publish':
+                case 'unpublish':
+                    $emptyValue = $this->modx->lexicon('notset');
+                    break;
+                case 'published':
+                    $emptyValue = $this->modx->lexicon('unpublished');
+                    break;
+                default:
+                    $emptyValue = $this->modx->lexicon('unknown');
+            }
+            $emptyValue = '(' . $emptyValue . ')';
+        }
+
+        return $this->formatDateTime($value, true, $emptyValue);
```

Comment:
I think we can lower the complexity of this fn, by doing: (I know i'm basically making suggestions to the code I wrote :D)
```suggestion
        if ($useStandardEmptyValue) {
            return $this->formatDateTime($value, true, null);
        } 
        
        switch ($whichDate) {
            case 'edited':
                $emptyValue = $this->modx->lexicon('unedited');
                break;
            case 'publish':
            case 'unpublish':
                $emptyValue = $this->modx->lexicon('notset');
                break;
            case 'published':
                $emptyValue = $this->modx->lexicon('unpublished');
                break;
            default:
                $emptyValue = $this->modx->lexicon('unknown');
        }
        $emptyValue = '(' . $emptyValue . ')';

        return $this->formatDateTime($value, true, $emptyValue);
```

Reply:
Done